### DISCUSSION TITLE
Multiple output for 1 epoch

### DISCUSSION THREAD

**Discussion by kfoynt** [snap-stanford/ogb#207] [MAG240M-LSC]
Hi, I've set 
`parser.add_argument('--epochs', type=int, default=1)`
in the baseline.
However, it seems from the output (see attached .txt) that train.fit does more than 1 epoch? 
Any ideas what are the extra calculations that are done? 

[slurm-19341562.txt](https://github.com/snap-stanford/ogb/files/6474112/slurm-19341562.txt)

**Comment by artur-deluca** [comment]
I think it's easier to visualize the slurm output as following. I'm experiencing the same issue...
```
Transferred dataset6/1224 [10:27<10:44,  1.35s/it]
Transferred files
Namespace(batch_size=1024, device='0', dropout=0.5, epochs=1, evaluate=False, hidden_channels=1024, in_memory=False, model='rgraphsage', sizes=[25, 15])
torch.float32
Global seed set to 42
GPU available: True, used: True
TPU available: None, using: 0 TPU cores
100%|██████████| 1218/1218 [51:48<00:00,  2.55s/it] 
100%|██████████| 1218/1218 [06:02<00:00,  3.36it/s]
100%|██████████| 1218/1218 [06:02<00:00,  4.23it/[artur Desktop]$ 
  2%|▏         | 26/1224 [00:54<52:19,  2.62s/it]

```

**Reply by weihua916** [reply]
We did not use Slurm but at least I got the parameter names printed before the training. Also progress bar looks like this:
```
Global seed set to 42
Epoch 0:  79%|█████████████████████████████████████        | 964/1223 [10:39<02:51,  1.51it/s, loss=1.22, v_num=0, val_acc=0.00146]
```
I feel this could be an issue of pytorch-lightning?

**Reply by kfoynt** [reply]
@weihua916 Are you using a conda environment?

**Comment by kfoynt** [comment]
Alright, that is weird. I used the baseline code on the small mag dataset. @weihua916 I am getting the output that you mentioned.
See the attached .txt. However, when I run the same code for the big dataset the output changes + lightning does not terminate at 1 epoch. 
[slurm-19384706.txt](https://github.com/snap-stanford/ogb/files/6478258/slurm-19384706.txt)

**Comment by kfoynt** [comment]
That was simply the data creation...

**Reply by weihua916** [reply]
I see. The pre-processing part :)
